tasks/task02/Area_of_a_triangle.py

```diff
@@ -0,0 +1,42 @@
+# main menu
+import math
+print("Welcome to the triangle area calculation tool.")
+while True:
+    print("\n","Menu:","1. Calculate triangle area by base and height","2. Calculate triangle area by 2 sides and angle between them","3. Exit", sep = "\n")
+    print("Enter menu item number: ", end = "")
+    mitem = input()
+    if mitem == "1":
+        print("Enter base and height (space separated): ", end = "")
+        while True:
+            inp = input().split(" ")
+            try:
+                base, height = [float(x) for x in inp]
```

Comment:
here you use list comprehension `[f(x) for x in seq]` to iterate and convert to float. it is fine, but `list` will allocate extra memory that is not needed here, the list is not reused anywhere.

for such cases [generator expression](https://www.python.org/dev/peps/pep-0289/) is a better choice. generators just generate a sequence of values on the fly, it is not stored in memory. more about generators vs lists: https://stackoverflow.com/questions/47789/generator-expressions-vs-list-comprehensions

that is, in this case it is better to write `base, height = (float(x) for x in inp)`.

bonus: there is a built-in function `map` that applies a function to every element of the sequence and produces a generator. so another way to write is:
```python
base, height = map(float, inp) # an equivalent of (float(x) for x in inp)
```